packages/pymedphys_deliverydata/src/pymedphys_deliverydata/object/exportable.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (C) 2019 Cancer Care Associates
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+import functools
+from typing import Union, Tuple
+
+
+from pymedphys_utilities.types import to_tuple
+from pymedphys_mudensity.mudensity import calc_mu_density
+
+from ..base import _DeliveryDataBase
+from ..dicom import (
+    delivery_data_to_dicom,
+    dicom_to_delivery_data,
+    gantry_tol_from_gantry_angles)
+from ..utilities import (
+    filter_out_irrelevant_control_points,
+    get_all_masked_delivery_data,
+    get_metersets_from_delivery_data)
+
+
+class DeliveryData(_DeliveryDataBase):
+    def __new__(cls, *args):
+        new_args = (
+            to_tuple(arg)
+            for arg in args
+        )
+        return super().__new__(cls, *new_args)
+
+    @classmethod
+    def from_delivery_data_base(cls, delivery_data_base):
+        if type(delivery_data_base) is type(cls):
+            return delivery_data_base
+
+        return cls(*delivery_data_base)
+
+    @classmethod
+    def from_dicom(cls, dataset):
+        return cls.from_delivery_data_base(
+            dicom_to_delivery_data(dataset))
+
+    @classmethod
+    def empty(cls):
+        return cls(
+            tuple(),
+            tuple(),
+            tuple(),
+            tuple((
+                tuple((
+                    tuple(),
+                    tuple()
+                )),
+            )),
+            tuple((
+                tuple(),
+                tuple()
+            ))
+        )
+
+    def to_dicom(self, template):
```

Comment:
Ah, you do have a `to_dicom()`.

Is it really necessary to maintain the function? (This is kinda what I mean by functionalised code, I now need to look elsewhere to find out what's going on here - in a different FILE even :/)